### PR TITLE
fix(cli): image publishing role doesn't have docker pull permissions

### LINKED ISSUE #14656: (cli): ImagePublishingRoleDefaultPolicy in bootstrap template should have docker pull permissions
Using a common docker asset as base image for other docker assets requires the image publishing role to have the `ecr:BatchGetImage`, `ecr:GetDownloadUrlForLayer` and `ecr:InitiateLayerUpload` permissions.

The comment here implies that it should be possible:
https://github.com/aws/aws-cdk/blob/0ea24e95939412765c0e09133a7793557f779c76/packages/cdk-assets/lib/private/handlers/container-images.ts#L34-L35

This is #6466 but for the new style stack synthesis.

### Use Case

Consider the following pattern to "reuse" a common docker asset as base image in other docker assets:
```ts
/**
 * A construct that creates a common docker asset that can be used
 * in multiple stacks as base image
 */
class CommonDockerAsset extends cdk.Construct {
  public readonly imageTag: string;

  constructor(scope: cdk.Construct, id: string) {
    super(scope, id);

    const asset = new assets.DockerImageAsset(this, 'Image', {
      directory: path.join(__dirname, '../common-docker'),
    });

    this.imageTag = asset.sourceHash;
  }
}
```

```docker
# Dockerfile for asset in StackA that should use the common docker
# asset as base image
ARG TAG
FROM 123456789012.dkr.ecr.eu-west-1.amazonaws.com/cdk-hnb659fds-container-assets-123456789012-eu-west-1:$TAG

COPY ...
```

```ts
/**
 * Use it in StackA and pass TAG as build arg
 */
class StackA extends TaggedStack {
  constructor(scope: cdk.Construct, id: string) {
    super(scope, id);

    const commonDockerAsset = new CommonDockerAsset(this, 'CommonDockerAsset');
    const taskDefinition = new ecs.FargateTaskDefinition(this, 'TaskDef');
    taskDefinition.addContainer('Container', {
      image: ecs.ContainerImage.fromAsset(path.join(__dirname, '../stack-docker'), {
        buildArgs: { TAG: commonDockerAsset.imageTag },
      }),
    });
  }
}
```

### Proposed Solution

Add the `ecr:BatchGetImage`, `ecr:GetDownloadUrlForLayer` and `ecr:InitiateLayerUpload` in the bootstrap template.

### Other

* [x] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by jogold**
Using a common docker asset as base image for other docker assets requires
the image publishing role to have the `ecr:BatchGetImage`, `ecr:GetDownloadUrlForLayer`
and `ecr:InitiateLayerUpload` permissions.

Closes #14656

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by jogold** [comment]
Does this need a test? If yes, where?

File: packages/aws-cdk/lib/api/bootstrap/bootstrap-template.yaml [hunk 1]
```diff
@@ -342,6 +342,9 @@ Resources:
               - ecr:BatchCheckLayerAvailability
               - ecr:DescribeRepositories
               - ecr:DescribeImages
+              - ecr:BatchGetImage
+              - ecr:GetDownloadUrlForLayer
+              - ecr:InitiateLayerUpload
```
Comment (rix0rrr) [inline packages/aws-cdk/lib/api/bootstrap/bootstrap-template.yaml:347]: `ecr:InitiateLayerUpload` is duplicated here.

Do you know why this wasn't detected earlier? Seems like a big miss if true...

**Review by rix0rrr — CHANGES_REQUESTED** [review]

**Review by rix0rrr — DISMISSED** [review]

File: packages/aws-cdk/lib/api/bootstrap/bootstrap-template.yaml (hunk 1 above)
Comment (jogold) [reply packages/aws-cdk/lib/api/bootstrap/bootstrap-template.yaml]: I really don't know... I only recently switched all my productions stack to the new style stack synthesis and came upon this.

While perfectly valid and working, I'm not sure that the pattern is commonly used...

Comment (rix0rrr) [reply packages/aws-cdk/lib/api/bootstrap/bootstrap-template.yaml]: I guess the issue only crops up when using an ECR image as a base, huh?

**Review by rix0rrr — APPROVED** [review]

File: packages/aws-cdk/lib/api/bootstrap/bootstrap-template.yaml (hunk 1 above)
Comment (jogold) [reply packages/aws-cdk/lib/api/bootstrap/bootstrap-template.yaml]: Yes

**Pull request merged**